lib/matplotlib/axes/_axes.py

```diff
@@ -2581,6 +2582,123 @@ def barh(self, y, width, height=0.8, left=None, *, align="center",
                            align=align, **kwargs)
         return patches
 
+    def bar_label(self, container, labels=None, *, fmt="%g", label_type="edge",
+                  padding=0, **kwargs):
+        """
+        Label a bar plot.
+
+        Adds labels to bars in the given `.BarContainer`.
+        You may need to adjust the axis limits to fit the labels.
+
+        Parameters
+        ----------
+        container : `.BarContainer`
+            Container with all the bars and optionally errorbars.
+
+        labels : array-like, optional
+            A list of label texts, that should be displayed. If not given, the
+            label texts will be the data values formatted with *fmt*.
+
+        fmt : str, default: '%g'
+            A format string for the label.
+
+        label_type : {'edge', 'center'}, default: 'edge'
+            The label type, Possible values:
+
+            - 'edge': label placed at the end-point of the bar segment, and the
+              value displayed will be the position of that end-point.
+            - 'center': label placed in the center of the bar segment, and the
+              value displayed will be the length of that segment.
+              (useful for stacked bars, i.e.
+              :doc:`/gallery/lines_bars_and_markers/bar_label_demo`)
+
+        padding : float, default: 0
+            Distance of label from the end of the bar.
+
+        **kwargs : passed through to `.Axes.annotate`.
+
+        Returns
+        -------
+        annotations
+            A list of `.Text` instances for the labels.
+        """
+
+        # want to know whether to put label on positive or negative direction
+        # cannot use np.sign here because it will return 0 if x == 0
+        def sign(x):
+            return 1 if x >= 0 else -1
+
+        cbook._check_in_list(['edge', 'center'], label_type=label_type)
+
+        bars = container.patches
+        errorbar = container.errorbar
+        orientation = container.orientation
+
+        if errorbar:
+            lines = errorbar.lines
+            barlinecols = lines[2]
+            barlinecol = barlinecols[0]
+            errs = barlinecol.get_segments()
+        else:
+            errs = []
+        lbls = [] if labels is None else labels
```

Comment:
Don't really need another variable.
```suggestion
        if labels is None:
            labels = []
```